src/click/core.py

```diff
@@ -1833,9 +1843,9 @@ def command(
 
         func: t.Optional[t.Callable] = None
 
-        if args and callable(args[0]):
-            func = args[0]
-            args = args[1:]
+        if len(args) == 1 and not kwargs and callable(args[0]):
+            (func,) = args
+            args = ()
```

Comment:
This was very loose - if you leave off the `()`, then there should be only one argument - the callable.